src/dappsExplorer/searchDappList.tsx

```diff
@@ -0,0 +1,111 @@
+import { get } from 'fast-levenshtein'
+import { DappV2WithCategoryNames } from 'src/dapps/types'
+
+const LEVENSHTEIN_THRESHOLD = 0.7 // levenshtein distance threshold
+
+enum ScoreValues {
+  High = 2.5,
+  Medium = 1.25,
+  Low = 0.625,
+}
+
+// Tokens are strings broken into lists of words by language agnostic regex
+const getTokens = (text: string, preserveCasing = false) => {
+  return preserveCasing
+    ? text.match(/[\p{L}]+/gu) ?? []
+    : text.toLowerCase().match(/[\p{L}]+/gu) ?? []
+}
+
+const getSimilarity = (a: string, b: string) => {
+  const maxLength = Math.max(a.length, b.length)
+  return 1 - get(a, b, { useCollator: true }) / maxLength
+}
+
+export const scoreDapp = (dapp: DappV2WithCategoryNames, searchTerm: string) => {
+  let score = 0
+  // Early return if there is no search term
+  if (searchTerm === '') return score
+
+  // Use a Map to keep take of the scores for each searchTerm
+  const scoreMap = new Map<string, number>()
+  const searchTokens = getTokens(searchTerm, true)
+  const searchTokensLower = getTokens(searchTerm)
+  const nameTokens = getTokens(dapp.name, true)
+  const descriptionTokens = getTokens(dapp.description)
+  // Lowercase categories as it was already an array of strings
+  const categoryTokens = dapp.categoryNames.map((category) => category.toLowerCase())
+
+  // Iterate through each word of and populate the scoreMap
+  // Use the searchTokens (Casing Preserved) to check for exact matches
+  for (const word of searchTokens) {
+    // Name scoring
+    if (nameTokens.some((token) => token.includes(word))) {
+      scoreMap.set(word, scoreMap.get(word) ?? 0 + ScoreValues.High)
+    } else {
+      for (const token of nameTokens) {
+        const similarity = getSimilarity(token, word)
+        if (similarity >= LEVENSHTEIN_THRESHOLD) {
+          scoreMap.set(word, (scoreMap.get(word) ?? 0) + ScoreValues.Medium)
+        }
+      }
+    }
+  }
+
+  // Use the lowercase search tokens to score the description and categories
+  for (const word of searchTokensLower) {
+    // Description scoring
+    if (descriptionTokens.some((token) => token.includes(word))) {
+      scoreMap.set(word, scoreMap.get(word) ?? 0 + ScoreValues.Medium)
+    } else {
+      for (const token of descriptionTokens) {
+        const similarity = getSimilarity(token, word)
+        if (similarity >= LEVENSHTEIN_THRESHOLD) {
+          scoreMap.set(word, (scoreMap.get(word) ?? 0) + ScoreValues.Low)
+        }
+      }
+    }
+
+    // Category scoring
```

Comment:
I'm still not sure if we want to add this to the scoring.

Reply:
I am worried without this we are hampering search and making search and filter a bit more difficult.

Reply:
Checked with @reedeereed and they seemed fine with this behavior.